### ISSUE TITLE
PUT resource policy allows assigning roles to non-`silo_user` users

### ISSUE THREAD

**Issue by david-crespo** [oxidecomputer/omicron#1246]
I've been using the system users from `/users` to test role assignment on orgs and projects because I can't list silo users yet (#1235). It just occurred to me that I should not be able to assign resource roles to system users! Here is the PUT body:

![image](https://user-images.githubusercontent.com/3612203/174885587-757421db-df29-4194-97b0-c424c7e8744b.png)

It seems that as long as I say `identity_type: "silo_user"`, the API does not validate that that is true.

https://github.com/oxidecomputer/omicron/blob/0588fac574963a9380313447bb90e1f2eda88cc5/nexus/src/db/datastore.rs#L3761-L3785

**Comment by davepacheco** [comment]
Yeah, this looks like a bug.  It _also_ seems like a separate bug if the resulting role assignment actually works.

**Comment by david-crespo** [comment]
I have not yet tested whether it actually works, but I am working on that.

**Comment by david-crespo** [comment]
Since @morlandi7 asked, I checked the code and can't see any evidence this has changed since being reported. Testing on colo rack:

```js
// trying to use the same ID twice. this 500s
(await oxide.policyUpdate({ body: { role_assignments: [
  { identity_type: "silo_group", identity_id: "7da1c977-ba19-4a26-bd68-3fb1463ab2c7", role_name: "admin" },
  { identity_type: "silo_group", identity_id: "7da1c977-ba19-4a26-bd68-3fb1463ab2c7", role_name: "admin" },
] } }))

// using a non-existent ID. this works just fine
(await oxide.policyUpdate({ body: { role_assignments: [
  { identity_type: "silo_group", identity_id: "7da1c977-ba19-4a26-bd68-3fb1463ab2c7", role_name: "admin" },
  { identity_type: "silo_group", identity_id: "00000000-0000-0000-0000-000000000000", role_name: "admin" },
] } })).data
```

So I'm thinking a better framing of this issue is that there is simply no validation on the IDs passed in, whether on their existence or their "type". The second post above also results in this amusing situation in the UI because we cannot find the ID in the list of users and groups.

<img width="1027" alt="image" src="https://github.com/user-attachments/assets/e961d4e8-669a-4fd7-815a-ea6667ad59fa">

**Comment by david-crespo** [comment]
Also, probably a separate issue, but a silo admin can nuke their own permissions by setting the policy to empty.

```js
await oxide.policyUpdate({ body: { role_assignments: [] } })
```

You can also do this the easy way: simply deleting the role assignment in the UI.

<img width="652" alt="image" src="https://github.com/user-attachments/assets/b9f26cf0-b255-44bc-89a1-d822065ff2c9">


At that point, I can load the empty access page, but pretty much any other page will blow up because I can't list projects or anything. The way we handle this is a matter of how the API handles UI errors, but currently we blow up with our most generic `Something went wrong` page.

<img width="1217" alt="image" src="https://github.com/user-attachments/assets/57152f38-cd9f-439c-b12b-ad7fbdfb4acc">
